components/SettingModal.tsx

```diff
@@ -84,20 +84,26 @@ const SettingModal = () => {
 
   return (
     <Container>
-      <IconButton onClick={openModal} >
-        <SettingsOutlined aria-label='open settings modal '
+      <IconButton onClick={openModal}>
+        <SettingsOutlined
+          aria-label="open settings modal "
           sx={{ fontSize: '24px', color: MuiVariants.NEUTRAL }}
         />
       </IconButton>
 
       {isOpen && (
         <ModalWrapper ref={ModalRef}>
           <ModalContent>
-            <CloseButton onClick={closeModal} onKeyDown={(e) => {
-              if(e.key == " " || e.key == "Enter" || e.key =="Return"){
-                closeModal()
-              }
-            }} tabIndex={0} aria-label="close settings modal">
+            <CloseButton
+              onClick={closeModal}
+              onKeyDown={(e: React.KeyboardEvent) => {
+                if (e.key == ' ' || e.key == 'Enter' || e.key == 'Return') {
```

Comment:
It was breaking the typing validation with TypeScript, fixing as well here 
"Error: Parameter 'e' implicitly has an 'any' type"